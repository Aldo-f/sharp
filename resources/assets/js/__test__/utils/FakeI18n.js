export default {
    install() {
        window.i18n = {};
    }
}