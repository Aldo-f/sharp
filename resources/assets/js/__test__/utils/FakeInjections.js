import Vue from 'vue';

const injectedComponents = ['$field', '$form'];

export default {
    provide() {
        return injectedComponents.reduce((injections, prop) => {
            injections[prop] = new Vue();
            return injections;
        }, {});
    }
}