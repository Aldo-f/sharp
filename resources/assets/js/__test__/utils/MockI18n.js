import Localization ,* as localizationFn from '../../mixins/Localization';

const mockLangImplementation = localeKey => `{{ ${localeKey} }}`;

export default {
    install() {
        window.i18n = {};
    },

    mockLangFunction() {
        localizationFn.lang = jest.fn(mockLangImplementation);
        Localization.methods.l = jest.fn(mockLangImplementation);
    }
}