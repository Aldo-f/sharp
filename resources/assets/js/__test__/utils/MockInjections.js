import Vue from 'vue';

const injectedComponents = {
    $field:{},
    $form:{
        data: () => ({
            errors: {}
        })
    },
    $tab:{}
};

export default {
    provide() {
        return Object.keys(injectedComponents).reduce((injections, compName) => {
            let Comp = Vue.extend(injectedComponents[compName]);
            injections[compName] = new Comp();
            return injections;
        }, {});
    }
}