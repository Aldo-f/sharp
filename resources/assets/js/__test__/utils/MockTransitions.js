export default {
    install() {
        window.getComputedStyle = jest.fn(() => ({
            animationDelay:'',
            animationDuration:'',
            transitionDelay:'',
            transitionDuration:''
        }));
    }
}