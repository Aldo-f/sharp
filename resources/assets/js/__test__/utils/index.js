export { default as MockInjections } from './MockInjections';
export { default as MockI18n } from './MockI18n';
export { default as QueryComponent } from './QueryComponent';
export { default as MockTransitions } from './MockTransitions';
export { mockSFC } from './mockSFC';

export function wait(delay) {
    return new Promise(resolve => {
        setTimeout(resolve, delay);
    });
}