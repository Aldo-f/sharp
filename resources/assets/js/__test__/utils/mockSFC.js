export function mockSFC (sfcModule, { template, render, ...extend }={}, mockOptions={}) {

    let { isJsFile } = mockOptions;

    if(isJsFile) {
        sfcModule.__oldDefault__ = { ...sfcModule.default }
    }

    Object.assign(sfcModule.default, {
        mixins: [],
        'extends': extend,
        template: template,
        render: !template ? render || function() { return this._v(`__MOCKED_SFC_${sfcModule.name}__\n`) } : null,
        watch: {},
        beforeCreate: ()=>{},
        created: ()=>{},
        beforeMount: ()=>{},
        mounted: ()=>{},
        beforeUpdate: ()=>{},
        updated: ()=>{},
        beforeDestroy: ()=>{},
        destroyed: ()=>{},
        data: ()=>({}),
        provide: {},
        inject: [],
    });

    //console.log(sfcModule);
    return sfcModule.default;
}

export function unmockSFC(sfcModule) {
    if(sfcModule.__oldDefault__) {
        sfcModule.default = sfcModule.__oldDefault__;
        return;
    }

    let { 'default':defaultObj, ...options } = sfcModule;
    sfcModule.default = options;
}