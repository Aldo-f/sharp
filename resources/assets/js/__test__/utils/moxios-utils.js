import moxios from 'moxios';

export function nextRequestFulfilled(config) {
    return new Promise(resolve => {
        moxios.wait(async () => {
            let request = moxios.requests.mostRecent();
            let response = await request.respondWith(config);
            resolve(response); // moxios Response object
        })
    });
}
