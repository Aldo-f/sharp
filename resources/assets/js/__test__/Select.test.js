import Vue from 'vue/dist/vue.common';
import Select from '../components/form/fields/Select.vue';

import { FakeI18n } from './utils';



describe('select-field',()=>{
    Vue.component('sharp-select', Select);
    Vue.use(FakeI18n);

    beforeEach(()=>{
        document.body.innerHTML = `
            <div id="app">
                <sharp-select :value="3" :multiple="multiple" :options="[{id:3,label:'AAA'}]" @input="inputEmitted($event)"></sharp-select>
            </div>
        `
    });

    it('can mount Select field', async () => {
        await createVm();
        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('clear on cross button clicked', async () => {
        let inputEmitted = jest.fn();

        await createVm({
            propsData: {
                multiple: false
            },
            methods: {
                inputEmitted
            }
        });

        let clearBtn = document.querySelector('.SharpSelect__clear-btn');
        clearBtn.dispatchEvent(new MouseEvent('mousedown'));

        expect(inputEmitted.mock.calls[0][0]).toBe(null);
    });

});

async function createVm(customOptions={}) {

    const vm = new Vue({
        el: '#app',
        mixins: [customOptions],

        props:['multiple']
    });

    await Vue.nextTick();

    return vm.$children[0];
}