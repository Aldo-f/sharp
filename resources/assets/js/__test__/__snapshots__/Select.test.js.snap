// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select-field can mount "read only" Select field as checkboxes list 1`] = `
<div id="app">
    <div class="SharpSelect SharpSelect--multiple">
        <div>
            <div class="SharpCheck"><label class="SharpCheck__label"><input type="checkbox" disabled="disabled" class="SharpCheck__input"> <span class="SharpCheck__appearance"><svg width="12" height="9" viewBox="0 0 12 9" fill-rule="evenodd" class="SharpCheck__checkmark"><path d="M4.1 6.1L1.4 3.4 0 4.9 4.1 9l7.6-7.6L10.3 0z"></path></svg></span> <span class="SharpChek__label-text">AAA</span></label></div>
            <div
                class="SharpCheck"><label class="SharpCheck__label"><input type="checkbox" disabled="disabled" class="SharpCheck__input"> <span class="SharpCheck__appearance"><svg width="12" height="9" viewBox="0 0 12 9" fill-rule="evenodd" class="SharpCheck__checkmark"><path d="M4.1 6.1L1.4 3.4 0 4.9 4.1 9l7.6-7.6L10.3 0z"></path></svg></span> <span class="SharpChek__label-text">BBB</span></label></div>
    </div>
</div>
</div>
`;

exports[`select-field can mount "read only" Select field as radios list 1`] = `
<div id="app">
    <div class="SharpSelect">
        <div>
            <div class="SharpSelect__radio-button-group"><span><input type="radio" id="select0" tabindex="0" disabled="disabled" name="select" class="SharpRadio"> <label for="select0" class="SharpRadio__label"><span class="SharpRadio__appearance"></span> AAA
                </label>
                </span><span><input type="radio" id="select1" tabindex="0" disabled="disabled" name="select" class="SharpRadio"> <label for="select1" class="SharpRadio__label"><span class="SharpRadio__appearance"></span> BBB
                </label>
                </span>
            </div>
        </div>
    </div>
</div>
`;

exports[`select-field can mount Select field 1`] = `
<div id="app">
    <div class="SharpSelect">
        <div tabindex="0" class="multiselect"><svg width="10" height="5" viewBox="0 0 10 5" fill-rule="evenodd" class="multiselect__select"><path d="M10 0L5 5 0 0z"></path></svg>
            <div class="multiselect__tags">
                <div class="multiselect__tags-wrap" style="display: none;"></div>
                <!---->
                <div class="multiselect__spinner" style="display: none;"></div>
                <!----><span class="multiselect__single">placeholder</span></div>
            <div class="multiselect__content-wrapper" style="max-height: 300px; display: none;">
                <ul class="multiselect__content" style="display: inline-block;">
                    <!---->
                    <li class="multiselect__element"><span data-select="Press enter to select" data-selected="Selected" data-deselect="Press enter to remove" class="multiselect__option multiselect__option--highlight"><span>AAA</span></span>
                        <!---->
                    </li>
                    <li class="multiselect__element"><span data-select="Press enter to select" data-selected="Selected" data-deselect="Press enter to remove" class="multiselect__option"><span>BBB</span></span>
                        <!---->
                    </li>
                    <li style="display: none;"><span class="multiselect__option">No elements found. Consider changing the search query.</span></li>
                </ul>
            </div>
        </div>
    </div>
</div>
`;

exports[`select-field can mount Select field as checkboxes list 1`] = `
<div id="app">
    <div class="SharpSelect SharpSelect--multiple">
        <div>
            <div class="SharpCheck"><label class="SharpCheck__label"><input type="checkbox" class="SharpCheck__input"> <span class="SharpCheck__appearance"><svg width="12" height="9" viewBox="0 0 12 9" fill-rule="evenodd" class="SharpCheck__checkmark"><path d="M4.1 6.1L1.4 3.4 0 4.9 4.1 9l7.6-7.6L10.3 0z"></path></svg></span> <span class="SharpChek__label-text">AAA</span></label></div>
            <div
                class="SharpCheck"><label class="SharpCheck__label"><input type="checkbox" class="SharpCheck__input"> <span class="SharpCheck__appearance"><svg width="12" height="9" viewBox="0 0 12 9" fill-rule="evenodd" class="SharpCheck__checkmark"><path d="M4.1 6.1L1.4 3.4 0 4.9 4.1 9l7.6-7.6L10.3 0z"></path></svg></span> <span class="SharpChek__label-text">BBB</span></label></div>
    </div>
</div>
</div>
`;

exports[`select-field can mount Select field as radios list 1`] = `
<div id="app">
    <div class="SharpSelect">
        <div>
            <div class="SharpSelect__radio-button-group"><span><input type="radio" id="select0" tabindex="0" name="select" class="SharpRadio"> <label for="select0" class="SharpRadio__label"><span class="SharpRadio__appearance"></span> AAA
                </label>
                </span><span><input type="radio" id="select1" tabindex="0" name="select" class="SharpRadio"> <label for="select1" class="SharpRadio__label"><span class="SharpRadio__appearance"></span> BBB
                </label>
                </span>
            </div>
        </div>
    </div>
</div>
`;
